### PR TITLE
[P-42005] Add "Add another" component for form wizard

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

Creates a new component to display fields as part of an "Add another" grouping.

### Issue tracking

- [P4-2005](https://dsdmoj.atlassian.net/browse/P4-2005)

## Screenshots

<kbd></kbd>

#### Automated testing

- [ ] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: common/helpers/field/set-optional-label.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    'fieldset.legend.text',
+    'fieldset.legend.html',
+  ]
+
+  if (isOptional) {
+    labelPaths.forEach(path => {
+      const key = get(translated, path)
+
+      if (key) {
+        set(translated, path, `${key} (${i18n.t('optional')})`)
```
Comment (solidgoldpig) [inline common/helpers/field/set-optional-label.js:23]: Comments as per https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/pull/737 ;)

File: common/components/add-another/template.njk [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }) %}
+    <div>
+      {{ govukButton({
+        html: t("actions::remove_item", {
+          name: itemName | lower,
+          index: loop.index
+        }),
+        classes: "govuk-button--secondary app-!-position-top-right",
+        name: "multiple-action",
+        value: "remove::" + params.name + "::" + loop.index0
```
Comment (solidgoldpig) [inline common/components/add-another/template.njk:31]: What happens if someone else removes an item and causes the index to get out of whack?

Would it be possible to use something to identify the actual blob of info to be zapped?

Comment (teneightfive) [reply common/components/add-another/template.njk:31]: Not sure what you mean? This loops over the items that exist in this session (that work is coming in #736)

So essentially this count is only ever for that user in that session.

Comment (solidgoldpig) [reply common/components/add-another/template.njk:31]: 👍

**Pull request merged**